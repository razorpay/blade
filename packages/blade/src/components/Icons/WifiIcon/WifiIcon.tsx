import { Svg, G, Path, Defs, ClipPath, Rect } from '../_Svg';
import type { IconComponent } from '..';
import useIconProps from '../useIconProps';

const WifiIcon: IconComponent = ({ size, color, ...styledProps }) => {
  const { height, width, iconColor } = useIconProps({ size, color });

  return (
    <Svg {...styledProps} width={width} height={height} viewBox="0 0 24 24" fill="none">
      <G clipPath="url(#clip0_59_173)">
        <Path
          d="M1.66126 10.7475C7.32983 5.75083 15.8302 5.75083 21.4988 10.7475C21.9131 11.1127 22.545 11.0729 22.9102 10.6586C23.2754 10.2443 23.2356 9.61238 22.8213 9.24718C16.3969 3.58427 6.76314 3.58427 0.338762 9.24718C-0.0755428 9.61238 -0.115352 10.2443 0.249847 10.6586C0.615045 11.0729 1.24696 11.1127 1.66126 10.7475Z"
          fill={iconColor}
        />
        <Path
          d="M5.2201 14.3158C8.92791 11.2274 14.3123 11.2274 18.0201 14.3158C18.4445 14.6692 19.075 14.6117 19.4285 14.1874C19.7819 13.763 19.7245 13.1325 19.3001 12.779C14.8507 9.07301 8.38948 9.07301 3.9401 12.779C3.51574 13.1325 3.45827 13.763 3.81173 14.1874C4.16519 14.6117 4.79574 14.6692 5.2201 14.3158Z"
          fill={iconColor}
        />
        <Path
          d="M14.4809 17.9225C12.7469 16.6905 10.4233 16.6905 8.68923 17.9225C8.239 18.2424 7.61472 18.1367 7.29486 17.6864C6.97499 17.2362 7.08067 16.6119 7.5309 16.2921C9.95853 14.5674 13.2116 14.5674 15.6392 16.2921C16.0895 16.6119 16.1951 17.2362 15.8753 17.6864C15.5554 18.1367 14.9311 18.2424 14.4809 17.9225Z"
          fill={iconColor}
        />
        <Path
          d="M12 22C12.5523 22 13 21.5523 13 21C13 20.4477 12.5523 20 12 20C11.4477 20 11 20.4477 11 21C11 21.5523 11.4477 22 12 22Z"
          fill={iconColor}
        />
      </G>
      <Defs>
        <ClipPath id="clip0_59_173">
          <Rect width="24" height="24" fill={iconColor} />
        </ClipPath>
      </Defs>
    </Svg>
  );
};

export default WifiIcon;
