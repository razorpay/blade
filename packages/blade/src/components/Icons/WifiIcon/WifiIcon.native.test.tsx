import WifiIcon from './';
import renderWithTheme from '~utils/testing/renderWithTheme.native';

describe('<WifiIcon />', () => {
  it('should render WifiIcon', () => {
    const renderTree = renderWithTheme(
      <WifiIcon color="feedback.icon.neutral.intense" size="large" />,
    ).toJSON();
    expect(renderTree).toMatchSnapshot();
  });
});
