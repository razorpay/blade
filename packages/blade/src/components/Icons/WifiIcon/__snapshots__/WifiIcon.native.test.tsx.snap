// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WifiIcon /> should render WifiIcon 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <RNSVGSvgView
    accessibilityElementsHidden={true}
    align="xMidYMid"
    bbHeight="20px"
    bbWidth="20px"
    data-blade-component="icon"
    fill="none"
    focusable={false}
    height="20px"
    importantForAccessibility="no-hide-descendants"
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        [
          {},
        ],
        {
          "flex": 0,
          "height": 20,
          "width": 20,
        },
      ]
    }
    vbHeight={24}
    vbWidth={24}
    width="20px"
  >
    <RNSVGGroup
      fill={null}
      propList={
        [
          "fill",
        ]
      }
    >
      <RNSVGGroup
        clipPath="clip0_59_173"
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
      >
        <RNSVGPath
          d="M1.66126 10.7475C7.32983 5.75083 15.8302 5.75083 21.4988 10.7475C21.9131 11.1127 22.545 11.0729 22.9102 10.6586C23.2754 10.2443 23.2356 9.61238 22.8213 9.24718C16.3969 3.58427 6.76314 3.58427 0.338762 9.24718C-0.0755428 9.61238 -0.115352 10.2443 0.249847 10.6586C0.615045 11.0729 1.24696 11.1127 1.66126 10.7475Z"
          fill={
            {
              "payload": 4280563015,
              "type": 0,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        />
        <RNSVGPath
          d="M5.2201 14.3158C8.92791 11.2274 14.3123 11.2274 18.0201 14.3158C18.4445 14.6692 19.075 14.6117 19.4285 14.1874C19.7819 13.763 19.7245 13.1325 19.3001 12.779C14.8507 9.07301 8.38948 9.07301 3.9401 12.779C3.51574 13.1325 3.45827 13.763 3.81173 14.1874C4.16519 14.6117 4.79574 14.6692 5.2201 14.3158Z"
          fill={
            {
              "payload": 4280563015,
              "type": 0,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        />
        <RNSVGPath
          d="M14.4809 17.9225C12.7469 16.6905 10.4233 16.6905 8.68923 17.9225C8.239 18.2424 7.61472 18.1367 7.29486 17.6864C6.97499 17.2362 7.08067 16.6119 7.5309 16.2921C9.95853 14.5674 13.2116 14.5674 15.6392 16.2921C16.0895 16.6119 16.1951 17.2362 15.8753 17.6864C15.5554 18.1367 14.9311 18.2424 14.4809 17.9225Z"
          fill={
            {
              "payload": 4280563015,
              "type": 0,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        />
        <RNSVGPath
          d="M12 22C12.5523 22 13 21.5523 13 21C13 20.4477 12.5523 20 12 20C11.4477 20 11 20.4477 11 21C11 21.5523 11.4477 22 12 22Z"
          fill={
            {
              "payload": 4280563015,
              "type": 0,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        />
      </RNSVGGroup>
      <RNSVGDefs>
        <RNSVGClipPath
          fill={
            {
              "payload": 4278190080,
              "type": 0,
            }
          }
          name="clip0_59_173"
        >
          <RNSVGRect
            fill={
              {
                "payload": 4280563015,
                "type": 0,
              }
            }
            height="24"
            propList={
              [
                "fill",
              ]
            }
            width="24"
            x={0}
            y={0}
          />
        </RNSVGClipPath>
      </RNSVGDefs>
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;
